### ISSUE TITLE
 Inference on Cytodark Dataset: Cell Mismatch Issue with Default Parameters

### ISSUE THREAD

**Issue by shubhankar-git** [digitalpathologybern/hover_next_inference#14]
Original image
![ID10_Aud_Cortex_Tursiops NISSL](https://github.com/user-attachments/assets/6d6de7b4-d274-4465-8c13-fcd7a0707ad3)

Predicted
![ID10_Aud_Cortex_Tursiops Predicted](https://github.com/user-attachments/assets/f61d9be0-ed70-4c14-9f41-4ccfabb9a184)

Ground Truth
![ID10_Aud_Cortex_Tursiops Ground Truth](https://github.com/user-attachments/assets/64c02a03-1f5d-46bb-b273-0f9a643541e3)

I ran inference on the [Cytodark dataset](https://zenodo.org/records/13694738) using the default parameters, but I noticed that a significant number of cells were mismatched between the predicted and ground truth results. I have attached the original Nissl image, the predicted output, and the ground truth image for reference.

Could you recommend hyperparameters or adjustments that can be applied to improve the cell segmentation results on this dataset? Any advice on tuning the model for better accuracy would be greatly appreciated.

**Comment by eliasbaumann** [comment]
Hey @shubhankar-git , thanks for opening this issue and reaching out! HoVer-NeXt was not trained on cytology data so its not validated for this kind of data. 
However, you could do a number of things:
1. From your question I could not get whether you trained the model on this dataset as well? If not, I would highly recommend doing that. Then you should optimize hyperparameters using the hyperparameter search script. Both training and hp-search are here [https://github.com/digitalpathologybern/hover_next_train](https://github.com/digitalpathologybern/hover_next_train) 
2. We are using object removal based on size in the post-processing. You can adjust those variables by changing the constants in [https://github.com/digitalpathologybern/hover_next_inference/blob/main/src/constants.py](https://github.com/digitalpathologybern/hover_next_inference/blob/main/src/constants.py). 

If I misunderstood what you are trying to do, please let me know as well!

**Comment by shubhankar-git** [comment]
Hey @eliasbaumann
Thanks for the response and the suggestions!

To clarify, I did not train the model on the Cytodark dataset; I used the pretrained `lizard_convnextv2_large` model for inference. While I am not planning to train the model myself, I will definitely try the hyperparameter tuning and post-processing adjustments you mentioned.

I'll look into the changes in the constants file and also experiment with the hyperparameter search script for optimization. I’ll keep you posted on how it goes and get back to you if I run into any issues.

Thanks again for your help!

**Comment by eliasbaumann** [comment]
You could start by setting [MAX_THRESHS_LIZARD ](https://github.com/digitalpathologybern/hover_next_inference/blob/f694ba93e4eca3299381d0a904c08ce3bc88004f/src/constants.py#L5C1-L5C20) to very large numbers, e.g. 100000.
But even then, the model will always try to find nuclei and not cells. And in the dataset you are using, the entire cell is annotated.

**Comment by eliasbaumann** [comment]
closing this issue for now - reopen this if you need further help with this specific topic.
